src/compiler/crystal/tools/tracing.cr

```diff
@@ -0,0 +1,337 @@
+require "colorize"
+require "string_pool"
+
+module Crystal
+  module Tracing
+    WORDS_DICTIONNARY = StringPool.new
+
+    class Values(T)
+      getter size : T = T.zero
+      getter sum : T = T.zero
+      getter min : T = T::MAX
+      getter max : T = T::MIN
+      @sum_square : T
+
+      def initialize
+        @sum_square = T.zero
+      end
+
+      def <<(value)
+        v = T.new(value)
+        @size += 1
+        @sum += v
+        @min = v if v < @min
+        @max = v if v > @max
+        @sum_square += v ** 2
+      end
+
+      def average
+        size > 0 ? sum / size : T.zero
+      end
+
+      def stddev
+        if size > 0
+          Math.sqrt((@sum_square / size) - (average ** 2))
+        else
+          T.zero
+        end
+      end
+
+      def humanize(io, value)
+        io << value.round(9)
+      end
+
+      def to_s(io : IO) : Nil
+        io << "[total="
+        humanize io, sum
+        io << " min="
+        humanize io, min
+        io << " max="
+        humanize io, max
+        io << " mean="
+        humanize io, average
+        io << " ±"
+        humanize io, stddev
+        io << ']'
+      end
+    end
+
+    class Durations < Values(Float64)
+      def humanize(io, value)
+        value = value.abs
+
+        if value >= 1
```

Comment:
The reason to not use `#humanize` is for performance?

Reply:
I just didn't know about these methods!